align/compiler/match_graph.py

```diff
@@ -70,7 +70,7 @@ def annotate(self):
             mapped_graph_list = self._mapped_graph_list(G1, circuit_name, self.pg )
             const_list = self.hier_graph_dict[circuit_name]['constraints']
             self.hier_graph_dict[circuit_name]["graph"] = self._reduce_graph(G1, circuit_name, mapped_graph_list, const_list)
-            #Removing single instances of instances
+            #Removing single instances of instances. TODO from sy: Generate or DoNotIdentify might be a single instance constraint. The logic below will fail.
```

Comment:
@kkunal1408 , there could be single instance constraints, why require len(instances) > 1?

Reply:
line # 78

Reply:
Need to be changed. as order constraint needs more than one and do_not_identify can have single instance.  Need two checks now.